docs/performance/proposal.md

```diff
@@ -0,0 +1,69 @@
+    - There is an AnalyzerRunner commandline tool checked into dotnet/roslyn that can be used to run analyzers and validate their performance. It needs to be run in a manual fashion.
+- [ASP.NET](https://github.com/aspnet/Benchmarks/blob/main/scenarios/README.md)
+    - Can be run on CI: **Yes**
+    - Can be run locally with a single script: **No**
+    - The ASP.NET team has written a took (crank) that allows them to run benchmarks on either their local machines or remote machines using a client/server model. This does not require the user to download the dotnet/performance repository manually to run scenarios from there. Users will need to manually setup/patch runtimes with their changes but can then run them against the real benchmarks from there.
+        - [Crank](https://github.com/dotnet/crank)
+        - [TechEmpower Benchmarks Power BI](https://msit.powerbi.com/view?r=eyJrIjoiYTZjMTk3YjEtMzQ3Yi00NTI5LTg5ZDItNmUyMGRlOTkwMGRlIiwidCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsImMiOjV9)
+- Runtime
+    - Can be run on CI: **Yes** CI runs require you to submit a PR against dotnet/performance
+    - Can be run locally with a single script: **No**
+    - The runtime team has a set of benchmarking guides that detail how to run the tests in dotnet/performance against local changes.
+        - [Benchmarking](https://github.com/dotnet/performance/blob/main/docs/benchmarking-workflow-dotnet-runtime.md)
+        - [Profiling](https://github.com/dotnet/performance/blob/main/docs/profiling-workflow-dotnet-runtime.md)
+
+## Proposed Workflow
+
+### Tests
+We will have two types of tests:
+
+#### Micro-Benchmarks
+A set of micro-benchmarks (written in BenchmarkDotnet) testing how much time analyzers spend computing result. Each new analyzer that ships in the SDK is expected to have a micro-benchmark that tests
+- code files that cause the analyzer to execute but not issue a a diagnostic.
+- code files that cause the analyzer to issue a diagnostic.
+
+These tests are expected to live in the dotnet/roslyn-analyzers repo to make local development simpler.
+
+#### End-to-End Tests
+An end-to-end compilation test that measures how long the build takes on a large real-world project (based off existing scenarios [here](https://github.com/dotnet/performance/blob/main/docs/sdk-scenarios.md#sdk-build-throughput-scenario)). This test will be run twice, once with all muti-core build features enabled (no `/m` is passed to msbuild, and `/parallel-` is passed to the compiler) and once with the SDK defaults enabled. The reason we will want a test with no parallelism is to make it easier to see the source of regressions. These test will not just measure how long it takes analyzers to execute but the entire SDK-based build process. It will need to collect an ETL file for investigation as well as the following metrics in a binlog file
+- How much time was spend in analysis (`/p:reportanalyzer=true`)
```

Comment:
💡 The output of this will likely need to be measured against a scenario-specific baseline. For example, IOperation analyzers should be evaluated against the performance of a baseline analyzer that constructs the entire IOperation tree but does not do any actual analysis of that tree.

Reply:
This is a good point. We may want to define different scenarios based on which APIs the analyzers touch. We may also want to add a scenario were we run no analyzers at all just so it will make narrowing down regressions easier. 

I still think the Pri0 scenario is testing analyzers that ship in the SDK with their default severities as this should be the 99.9% use case in the wild.